### ISSUE TITLE
3D Videos are broken on the new VideoPlayerInterface

### ISSUE THREAD

**Issue by hazre** [Yellow-Dog-Man/Resonite-Issues#39]
### Describe the bug?

3D videos do not work properly, they only render in one eye. They work fine in LegacyVideoPlayer works fine.

### To Reproduce

Try to play a 3D video.

### Expected behavior

For 3D videos to be actually 3D. 

### Resonite Version Number

Any

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Quest Pro

### Reporters

@hazre

**Comment by Rellik66** [comment]
Additional details:
3D videos are primarily broken when spawning new players, old players seem to work fine.
This is most likely due to old players using unlit material with righteye adjustments, but the new players use projection360material instead.
I briefly tried playing with the offsets on projection360, but I couldn't figure it out.

**Comment by Ardescar** [comment]
So this is then a Content Bug / Issue @shiftyscales @RyuviTheViali 

Projektion360Material: The RightEyeTexture Offset and Scale are swapped in the inspector compared to UnlitMaterial. 

It needs to look like this. 
![grafik](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/63201013/158a91a2-8c0e-4009-aa3f-d8b7cafa1656)

And not like this.
![grafik](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/63201013/7a7a255c-3887-4831-961e-c4bf169a5237)

Currently you cant watch a 3d Movie without changing this manually.
![grafik](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/63201013/ef3401d6-55e0-4d7a-896d-659bf5b3ed7f)

**Comment by Frooxius** [comment]
Can you provide a sample 3D video that we can replicate this with?

**Comment by shadowpanther** [comment]
Basically any side-by-side or top-and-bottom video would probably have this issue. I also had this issue the other night, though I can't share the exact videos I tried.

I'd try to look into texture settings of 3D videos and I'll let you know if I find something that could be easily fixed.

**Comment by Frooxius** [comment]
Generally I'd recommend including it in the replication steps, otherwise it adds more complexity on our end.

It makes the decision between:
- Oh video URL, I'll just copy & paste and verify, which takes a few seconds to do
- Do I spend several minutes searching for some 3D video somewhere and then checking if it matches the format and style of the one reported or do I defer this for some other time and click to another issue to look into instead?

If you can't share the video (or any asset), it helps to find if you can find one that you can that also replicates the issue in the same way.

**Comment by shadowpanther** [comment]
Yeah, sorry about that. Should've provided the video too. Here's a folder link:
resrec:///U-Shadow-Panther/R-428E052000FE9195EC8B926FA5D16F70DE5EAB5F1D4E43FEAD3203D340296E1C

Import the raw file as Stereo Video < SBS left-right.
There's also the pre-imported video in the same folder.

This is what the original (non-stereofied) video frame looks like - two full frames stacked side by side in one:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/675104/72e009c0-1cf1-407f-866c-00c3c98ab020)

This is my attempt to shoot a 3D photo with the default camera (notice that the aspect ratio is wrong, the camera taking two left-eye pictures is a separate issue):
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/675104/18a67356-6014-49db-8d0b-b31bddcdc7b5)

This is the video material used in the video player when imported as a stereo video - it uses a Projection360 for some reason:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/675104/d3041ea6-03f7-4ef9-b0af-ecfe0f45e380)

And this is the stereo _picture_ (Quad with Unlit) where the texture ref is replaced with the VideoTextureProvider - normal aspect ratio, both left and right eye display properly:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/675104/52e28e3a-100f-4d1c-ad8c-d5164274cdef)

**Comment by shadowpanther** [comment]
And fixed it (kinda, the aspect ratio is still wrong):
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/675104/8605ffa4-7616-45b3-a1aa-60867b6de729)

The TextureOffsets should be [0,0] and [0.5,0], not [0.5,1]

**Comment by shadowpanther** [comment]
Also, when importing the video as "180 Video" and "180 Stereo Video", the projection sphere is rotated backwards, so that the projected half is only visible if you are looking from the player's back side (the one that doesn't have controls and flat video).

**Comment by hazre** [comment]
> @hazre the content of resonite is what this means - not the content that you play. As we are able to correct the issue by changing some values in the player itself its higly possible that this is a content issue.
> 
> **Content Issue - For issues with tutorial/home worlds, default facets, video and audio players.**

So don't you think default values being problematic here is the issue? If someone is trying to create a video player from scratch in isolation, are they just supposed to just play around the values until a basic functionality like playing 3D videos in a VR application finally works?

**Comment by Ardescar** [comment]
It would have been good tho if Ryuvi or Aegis could have looked into.

**Comment by Frooxius** [comment]
@RyuviTheViali or @BlueCyro are you able to take this one?

**Comment by Frooxius** [comment]
@RyuviTheViali Has this been handled at some point?
